lib/build.js

```diff
@@ -152,24 +154,36 @@ class BuildModel extends BaseModel {
     start() {
         // Make sure that a pipeline and job is associated with the build
         return this.job
-            .then(job =>
-                job.pipeline.then(pipeline =>
+            .then((job) => {
+                let buildTimeout = DEFAULT_BUILD_TIMEOUT;
+
+                // buildTimeout is only set in executor-k8s or executor-k8s-vm
+                if (this[executor].kubernetes && this[executor].kubernetes.buildTimeout) {
+                    buildTimeout = this[executor].kubernetes.buildTimeout;
+                }
+
+                const annotations = job.permutations[0].annotations;
+                const jwtExpiresSec = annotations[ANNOTATION_BUILD_TIMEOUT]
+                    ? annotations[ANNOTATION_BUILD_TIMEOUT] * 60
```

Comment:
Should there be a check here to make sure the `annotations[ANNOTATION_BUILD_TIMEOUT]` is less than the `MAX_BUILD_TIMEOUT`? https://github.com/screwdriver-cd/executor-k8s-vm/blob/master/index.js#L204-L206

Reply:
This `jwtExpiresSec` value is only used in `tokenGen(generateToken)` method, and I added the check that it is less than `max build timeout` in `generateToken` method.